### PR TITLE
Simpler, stricter and complete precedence and associativity tests.

### LINKED ISSUE #38: Unit test operator precedence

### PR DESCRIPTION
**Pull request by Renelvon**
Fix #38.
### Parse test:
- Complete unit tests for operator precedence.
- Complete unit tests for operator associativity.
- Add helper function to assert parsing failure; use it.

The precedence tests are written vis a vis the spec (pg. 14). To avoid quadratic blow-up of testcases, only operators of adjacent levels of precedence are (extensively) tested, with a few exceptions. The tests are meant to guard against accidental errors/omissions, not deliberate messing with the parser.

### CODE REVIEW

File: tests/test_parser.py [hunk 1]
```diff
@@ -421,113 +427,391 @@ def _assert_non_equivalent(self, expr1, expr2=None, start="expr"):
...
+            ("-1 ** 2", "(-1) ** 2"),
+            ("1 ** -2", "1 ** (-2)"),
+
+            ("+.1 ** 2", "(+.1) ** 2"),
+            ("1 ** +.2", "1 ** (+.2)"),
+
+            ("-.1 ** 2", "(-.1) ** 2"),
+            ("1 ** -.2", "1 ** (-.2)"),
+
+            ("not true ** 2", "(not true) ** 2"),
```
Comment (dionyziz) [inline tests/test_parser.py:486]: `not true` returns a boolean. Is it legal to raise it to a power? If not, then this test is esoteric and has more knowledge about the parser than it should.

File: tests/test_parser.py [hunk 2]
```diff
@@ -421,113 +427,391 @@ def _assert_non_equivalent(self, expr1, expr2=None, start="expr"):
...
         ))
 
-    def test_precedence_rest(self):
+    def test_precedence_bor_assign(self):
         self._assert_equivalent((
-            # function and constructor calls
-            ("f 1 + 2", "(f 1) + 2"),
-            ("F 1 + 2", "(F 1) + 2"),
+            ("a := b || c", "a := (b || c)"),
+            ("a || b := c", "(a || b) := c"),
```
Comment (dionyziz) [inline tests/test_parser.py:669]: This is illegal in the language.

Comment (Renelvon) [inline tests/test_parser.py:669]: It is legal syntax; it is illegal semantics.

File: tests/test_parser.py (hunk 1 above)
Comment (Renelvon) [inline tests/test_parser.py:486]: It is legal syntax, so I have to test it.

**Pull request merged**